runtime.c

```diff
@@ -385,6 +392,38 @@ main (int argc, char *argv[])
         exit(0);
     }
 
+    if (arg && strcmp(arg,"appimage-portable-home")==0) {
+        char portable_home[2048];
```

Comment:
Looks like this is hoping that the argument won't ever exceed this size. This should be replaced with a `malloc()`/`calloc()` based solution, properly allocating the size required for `portable_home` (easy to calculate with `strlen(appimage_path) + strlen(".home")`).

Reply:
As said before, I used the same value used for `portable_home_dir` and `portable_config_dir`.

However, I will replace all those values with `PATH_MAX`, which is the same value `appimage_path` uses.